src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameOfT.cs

```diff
@@ -158,6 +154,8 @@ public class Frame<TContext> : Frame
             {
                 try
                 {
+                    await TryProduceEnd();
```

Comment:
This might actually break server shutdown as written. We might return an empty 200 response when before we would close the connection. Closing the connection immediately in the middle of reading the start line or headers may have been wrong, but returning a 200 without calling application code would be worse.